src/PollySandbox/ResilienceStrategyStore.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Copyright (c) Martin Costello, 2023. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Globalization;
+using System.Net;
+using System.Threading.RateLimiting;
+using Microsoft.Extensions.Caching.Memory;
+using Polly;
+using Polly.CircuitBreaker;
+using Polly.RateLimiting;
+using Polly.Retry;
+using Polly.Timeout;
+using Refit;
+
+namespace PollySandbox;
+
+public partial class ResilienceStrategyStore
```

Comment:
I think it might be possible to simplify or to completely delete this class if you leverage the `AddResilienceStrategy` extension on top of `IServiceCollection`. 

It takes care of caching and invalidation of cached strategies. This PR can  serve as a reference:
https://github.com/App-vNext/Polly/pull/1366

Also, you could move of the code from here to `PollyExtensions`.